### PR TITLE
fix: M-01 Change of collateral could result in unintended bond value

### PR DESCRIPTION
**Pull request by pemulis**
Signed-off-by: John Shutt <pemulis@users.noreply.github.com>

**Motivation**

Updating the bond and collateral in the same function prevents the possibility of a gap between updating the collateral type and updating the bond, which could result in a proposal with an incorrect bond size.

**Summary**

Creates a `setCollateralAndBond` function that sets the collateral and bond simultaneously and remove the two separate functions.

**Details**

This PR also adds events for `SetBond` and `SetCollateral` in line with the fixes for M-02.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/core/contracts/zodiac/OptimisticGovernor.sol [hunk 1]
```diff
@@ -113,22 +117,20 @@ contract OptimisticGovernor is Module, Lockable {
...
+     * @notice Sets the collateral and bond amount for proposals.
      * @param _collateral token that will be used for all bonds for the contract.
+     * @param _bond amount of the bond token that will need to be paid for future proposals.
      */
-    function setCollateral(IERC20 _collateral) public onlyOwner {
+    function setCollateralAndBond(IERC20 _collateral, uint256 _bond) public onlyOwner {
         // ERC20 token to be used as collateral (must be approved by UMA Store contract).
         require(_getCollateralWhitelist().isOnWhitelist(address(_collateral)), "bond token not supported");
         collateral = _collateral;
+        emit SetCollateral(_collateral);
```
Comment (chrismaree) [inline packages/core/contracts/zodiac/OptimisticGovernor.sol:128]: nit: also emit `_bond`

File: packages/core/contracts/zodiac/OptimisticGovernor.sol [hunk 2]
```diff
@@ -113,22 +117,20 @@ contract OptimisticGovernor is Module, Lockable {
...
+
+        // Value of the bond required for proposals, in addition to the final fee. A bond of zero is
+        // acceptable, in which case the Optimistic Oracle will require the final fee as the bond.
+        bond = _bond;
+        emit SetBond(_bond);
```
Comment (chrismaree) [inline packages/core/contracts/zodiac/OptimisticGovernor.sol:133]: why use two seperate events? prehaps make an event `SetBond` with `collateral` and `bond` amount

File: packages/core/contracts/zodiac/OptimisticGovernor.sol [hunk 3]
```diff
@@ -113,22 +117,20 @@ contract OptimisticGovernor is Module, Lockable {
...
-    }
-
-    /**
-     * @notice Sets the collateral token (and bond token) for future proposals.
+     * @notice Sets the collateral and bond amount for proposals.
      * @param _collateral token that will be used for all bonds for the contract.
+     * @param _bond amount of the bond token that will need to be paid for future proposals.
      */
-    function setCollateral(IERC20 _collateral) public onlyOwner {
+    function setCollateralAndBond(IERC20 _collateral, uint256 _bond) public onlyOwner {
```
Comment (chrismaree) [inline packages/core/contracts/zodiac/OptimisticGovernor.sol:124]: a better variable name might be `_bondAmount`

File: packages/core/contracts/zodiac/OptimisticGovernor.sol (hunk 2 above)
Comment (mrice32) [reply packages/core/contracts/zodiac/OptimisticGovernor.sol:133]: +1

File: packages/core/contracts/zodiac/OptimisticGovernor.sol (hunk 3 above)
Comment (mrice32) [reply packages/core/contracts/zodiac/OptimisticGovernor.sol:124]: +1

**Review by mrice32 — COMMENTED** [review]
LGTM once @chrismaree comments are resolved!

File: packages/core/contracts/zodiac/OptimisticGovernor.sol (hunk 1 above)
Comment (nicholaspai) [inline packages/core/contracts/zodiac/OptimisticGovernor.sol:128]: nit: should emit all events at the end at the same time, but I also think you should consoldiate these events into one. There's no advantage to emitting an event here, changing state after, then emitting another

**Review by nicholaspai — APPROVED** [review]
LGTM once you resolve comments

**Comment by pemulis** [comment]
@mrice32 @nicholaspai running into the same issue where the automated tests are failing for reasons unrelated to this PR but I don't have the ability to re-run the tests. Could I be added to the group that can trigger a re-run of the test suite? @evaldofelipe

**Pull request merged**